File: python_team2/common/optimizer.py
# -*- coding: utf-8

# 1.3.6 重みの更新

class SGD:
    def __init__(self, lr=0.01):
        self.lr = lr

    def update(self, params, grads):
        for i in range(len(params)):
            params[i] -= self.lr * grads[i]

File: ruby/lib/optimizer.rb
# Stochastic Gradient Descent
class SGD
  def initialize(lr = 0.01)
    @lr = lr
  end

  def update(params, grads)
    params.length.times do |i|
      params[i].inplace - @lr * grads[i]
    end
  end
end
